### PR TITLE
Add Ueberauth

### LINKED ISSUE #25: Add ueberauth
See https://github.com/ueberauth/ueberauth

We will probably want support for [Facebook](https://github.com/ueberauth/ueberauth_facebook) and [Google](https://github.com/ueberauth/ueberauth_google)

[Authentication via OAuth](https://www.dailydrip.com/topics/elixir/drips/authentication-via-oauth) from Daily Drip

- [ ] Facebook
- [x] Google
- [x] Identity (local account)

### PR DESCRIPTION
**Pull request by Serneum**
Converted the login page to use Ueberauth's Identity strategy
Added the ability to log in with Facebook

TODO:
- [x] Add Google auth
- [x] Add links/buttons for Facebook/Google auth
- [ ] Tests?
- [ ] Facebook flow

**Only review the last three commits**

Closes #25 

login

### CODE REVIEW

**Comment by daveshah** [comment]
Ahhh! You beat me to this! :) I've been pretty busy this past week but glad you added it. 
We should probably talk through `Add Google auth` a bit. I've added both Facebook and Google in the past through Ueberauth and there were some interesting problems there.

**Comment by Serneum** [comment]
Noted. I just assumed I'd be able to look at the provider and figure out what to do from there, kinda like I already did with `identity` vs `facebook`. I'd also like to talk about how to test this. Ideally I'd stub it out somehow and just return the data I want

**Comment by Serneum** [comment]
@daveshah I was able to get Google working pretty easily. Take a look at the commit.

Note:
For both Facebook and Google, I created two applications: one for testing and one for prod. Only the testing credentials are committed for ease of development. Production credentials are already set in Heroku

**Comment by daveshah** [comment]
@Serneum the problems I've had were understanding what account to link to. According to the [Facebook docs](https://developers.facebook.com/docs/facebook-login/permissions/#reference-email), email isn't a guarantee, even if you request it. This can be problematic if you're not careful about the account creation workflow.

File: lib/agoneum/account/session.ex [hunk 1]
```diff
@@ -3,16 +3,34 @@ defmodule Agoneum.Account.Session do
...
 
     case check_password(user, password) do
       true -> {:ok, user}
       _    -> {:error, :invalid_credentials}
     end
   end
-  def authenticate(_), do: {:error, :invalid_credentials}
+  defp authenticate(:identity, _, _), do: {:error, :invalid_credentials}
+  defp authenticate(_, _, %Ueberauth.Auth.Info{email: email, name: name}) do
+    case Account.get_user_by_email(email) do
```
Comment (daveshah) [inline lib/agoneum/account/session.ex:20]: This might not be good enough. See [this](https://github.com/ElixirCLE/agoneum/pull/27#issuecomment-316958953)

**Comment by Serneum** [comment]
I'm thinking we just return an error if they used a phone number. Our sign up is relatively simple (for now) and only requires an email, name, and a password. If Facebook doesn't work, they can just do the manual sign up or use Google. We can address it if we ever get a large enough user base that it matters

**Pull request merged**